app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java

```diff
@@ -39,4 +49,44 @@ public void setFrom(Place from) {
         this.from = from;
     }
 
+    public void setShuttles() {
+        shuttles = appDB.shuttleDao().getAll();
+    }
+
+    public List<Shuttle> getShuttles() {
+        String campusFrom = "";
+        String campusTo = "";
+        if (getFrom() != null && getTo() != null) {
+            if (from.getClass() == Building.class) {
+                campusFrom = ((Building) from).getCampus();
+            } else if (to.getClass() == Building.class) {
+                campusTo = ((Building) to).getCampus();
+            } else if (from.getClass() == Floor.class) {
+                campusFrom = ((Floor) from).getCampus();
+            } else if(to.getClass() == Floor.class) {
+                campusTo = ((Floor) to).getCampus();
+            }
+            if (campusFrom.compareTo(campusTo) == 0) {
+                return null;
+            }
```

Comment:
Can you elaborate a bit more about this if else statements and what their purpose is ?

Reply:
Because the from and to are places, so I can't get the campus since a place doesn't have campus. So I am checking if it is an instance of Floor or Building, then I am casting accordingly to be able to get the campus.

Reply:
what about if the place is a room @CarenRizk ?

Reply:
I see that floor and building both have a campus. we should also consider this for room. i think it makes sense to add campus to place

Reply:
I agree with thomas of removing the campus attribute from all the subclasses and leave it only in the place class. However, we will need to modify the json file of the rooms accordingly to add the campus key and values in there for each room